becquerel/core/fitting.py

```diff
@@ -0,0 +1,816 @@
+from __future__ import print_function
+import warnings
+import inspect
+import numpy as np
+import pandas as pd
+import scipy.special
+from lmfit.model import Model
+import matplotlib.pyplot as plt
+from matplotlib.gridspec import GridSpec
+from matplotlib.font_manager import FontProperties
+from .utils import isstring
+
+FWHM_SIG_RATIO = np.sqrt(8*np.log(2))  # 2.35482
+SQRT_TWO = np.sqrt(2) #1.414213562
+
+class FittingError(Exception):
+    """Exception raised by Fitters."""
+    pass
+
+
+class FittingWarning(UserWarning):
+    """Warning raised by Fitters."""
+
+
+# -----------------------------------------------------------------------------
+# Base functions
+# TODO: Should these be replaced with lmfit.models???
+# -----------------------------------------------------------------------------
+
+
+def constant(x, c):
+    return np.ones_like(x) * c
+
+
+def line(x, m, b):
+    return m * x + b
+
+
+def gauss(x, amp, mu, sigma):
+    amp = np.abs(amp)
+    mu = np.abs(mu)
+    sigma = np.abs(sigma)
+    return amp / np.sqrt(2. * sigma ** 2. * np.pi) * \
```

Comment:
```suggestion
    return amp / sigma / np.sqrt(2. * np.pi)` * \
```

Reply:
Agreed. Is there a strong reason for `abs`ing all the inputs?

Reply:
I think that is so that the optimization can't make them negative.

Reply:
I suppose in general one could want to fit a peak with a negative mean (it's only a coordinate shift, after all) or perhaps even a trough with negative amplitude. We could probably revisit this later though. There might be a way to bound the search similar to providing initial guesses in `lmfit`, rather than bounding the fit functions themselves.

Reply:
See also @markbandstra's comment above on parameter bounds

Reply:
Yeah forcing the mean to be positive doesn't make any sense. Negative amplitudes I am not sure, often you don't want that, but there are ways to enforce that in the fitting routine